src/czbenchmarks/datasets/utils.py

```diff
@@ -12,17 +13,137 @@
 log = logging.getLogger(__name__)
 
 
+def _generate_dataset_name_from_path(path: str) -> str:
+    """
+    Generate a unique dataset name from a file path.
+    
+    Args:
+        path (str): File path to convert to dataset name
+        
+    Returns:
+        str: A valid dataset name derived from the path
+    """
+    # Extract filename without extension
+    filename = os.path.splitext(os.path.basename(path))[0]
+    # Replace invalid characters with underscores
+    safe_name = "".join(c if c.isalnum() else "_" for c in filename)
+    # Remove consecutive underscores and strip trailing ones
+    safe_name = "_".join(filter(None, safe_name.split("_")))
+    return f"user_dataset_{safe_name}"
+
+
+def _handle_dataset_path(path: str) -> str:
+    """
+    Handle dataset path for both local files (file://) and remote files (s3://).
+    
+    Args:
+        path (str): Dataset path, either local (file://) or remote (s3://)
+        
+    Returns:
+        str: Local file path to the dataset
+        
+    Raises:
+        FileNotFoundError: If local file doesn't exist
+        ValueError: If file:// path is invalid
+    """
+    if path.startswith("file://"):
+        # Handle local file
+        local_path = path[7:]  # Remove "file://" prefix
+        
+        # Convert to absolute path and expand user directory
+        local_path = os.path.expanduser(local_path)
+        local_path = os.path.abspath(local_path)
+        
+        # Check if file exists
+        if not os.path.exists(local_path):
+            raise FileNotFoundError(f"Local dataset file not found: {local_path}")
+        
+        log.info(f"Using local dataset file: {local_path}")
+        return local_path
+    else:
+        # Handle remote file (existing functionality)
+        return download_file_from_remote(path)
+
+
+def load_local_dataset(
```

Comment:
Might it be simpler to keep load_local_dataset() and load_dataset() completely independent? The local dataset function could avoid dealing with OmegaConf objects entirely as it would just directly instantiate the provided Dataset class (if provided as a Dataset class type param).  And then load_dataset() remains unchanged and is used exclusively for the remote dataset case and the mutually exclusive params it has wouldn't be needed (having both in one method confuses usage, IMO). Unless `hydra.utils.instantiate` implementation is providing something particularly useful?